src/plugindata.h

```diff
@@ -60,9 +60,9 @@ G_BEGIN_DECLS
  */
 #define GEANY_API_VERSION 239
 
-
-/* when gtk2 and gtk3 were both supported at the same time this was used
- * to prevent loading gtk2 plugins with gtk3 Geany or vice versa. */
+/* hack to have a different ABI when built with different GTK major versions
+ * because loading plugins linked to a different one leads to crashes.
+ * Only GTK3 is currently supported. */
```

Comment:
Again here I think it's worth keeping the idea itself for future use.  But I'm open to any rewording if you want.

Reply:
Why not make the "hack" official and extendable for the next major incompatibility be it GTK4 or whatever, but for now only one major ABI is available.

Reply:
Yeah maybe we could promote that as non-hack.  For other GTK versions, it's just a matter of shifting by an additional 8 or so.  That could be something like this:
```c
#define GEANY_ABI_SHIFT (8 * (GTK_MAJOR_VERSION - 2))
```
Though, I figure the shift was kind of misguided, because it doesn't make much sense to actually shift, we could just offset, and we'd loose a whole lot less values.

Maybe we could change this to something like this instead:
```c
/* 18360 is (72<<8) which was the actual value of the ABI while it used
 * shifting, minus 72 that is still added as the base ABI version. This
 * was with GTK3; with each newer version we move the offset by an
 * additional 1024 of available ABI range */
#define GEANY_ABI_OFFSET (18360 + 1024 * (GTK_MAJOR_VERSION - 3))
#define GEANY_ABI_VERSION (72 + GEANY_ABI_OFFSET)

Reply:
> we could just offset, and we'd loose a whole lot less values.

Agreed, IIUC your suggestion is allowing 1024 per version which should be enough on past experience.

Reply:
Yes.  We could actually make that smaller from past experience but I figure we've got a whole bunch of values left above 18432 so that we can spare 1024 ranges.  That would get us to GTK 49 before reaching a mere 16 bits worth of values.

Reply:
I agree with this, this should be changed (but can be done separately right?)